flycheck.el

```diff
@@ -6920,6 +6922,35 @@ See URL `http://www.erlang.org/'."
    (error line-start (file-name) ":" line ": " (message) line-end))
   :modes erlang-mode)
 
+(defun flycheck-rebar3-project-root (&optional _checker)
+  "Return directory where =rebar.config= is located."
+  (let* ((rebarconfig (locate-dominating-file buffer-file-name "rebar.config"))
+         (project-dir (if rebarconfig
+                          (file-name-as-directory
+                           (substring rebarconfig
+                                      0
+                                      (string-match
+                                       "\\(/_build/[^/]+/lib\\|/_checkouts\\)"
+                                       rebarconfig)))
+                        rebarconfig)))
+    project-dir))
+
+(flycheck-define-checker erlang-rebar3
+  "An Erlang syntax checker using the rebar3 build tool."
+  :command ("rebar3" "compile")
+  :error-parser
+  (lambda (output checker buffer)
+    (flycheck-parse-with-patterns (ansi-color-filter-apply output) checker buffer))
```

Comment:
Please add a little comment here that explains why we need to filter colours manually, and include a link to this pull request, so that we can easily find the discussion again when we look at this code in future.